services/skus/controllers.go

```diff
@@ -943,54 +947,79 @@ func MerchantTransactions(service *Service) handlers.AppHandler {
 	})
 }
 
-func VerifyCredentialV2(service *Service) handlers.AppHandler {
+func handleVerifyCredV2(svc *Service, valid *validator.Validate) handlers.AppHandler {
 	return func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
 		ctx := r.Context()
 
-		l := logging.Logger(ctx, "skus").With().Str("func", "VerifyCredentialV2").Logger()
+		lg := logging.Logger(ctx, "skus").With().Str("func", "handleVerifyCredV2").Logger()
+
+		data, err := io.ReadAll(io.LimitReader(r.Body, reqBodyLimit10MB))
+		if err != nil {
+			lg.Warn().Err(err).Msg("failed to read body")
+
+			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+		}
+
+		req, err := parseVerifyCredRequestV2(data)
+		if err != nil {
+			lg.Warn().Err(err).Msg("failed to deserialize request")
 
-		req := &VerifyCredentialRequestV2{}
-		if err := inputs.DecodeAndValidateReader(ctx, req, r.Body); err != nil {
-			l.Error().Err(err).Msg("failed to read request")
 			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
 		}
 
-		appErr := service.verifyCredential(ctx, req, w)
-		if appErr != nil {
-			l.Error().Err(appErr).Msg("failed to verify credential")
+		if err := validateVerifyCredRequestV2(valid, req); err != nil {
+			verrs, ok := collectValidationErrors(err)
+			if !ok {
+				return handlers.ValidationError("request", map[string]interface{}{"request-body": err.Error()})
```

Comment:
question: will this change format of the response to the client?

Reply:
It will, I guess. Thought about this, and was choosing between a better representation and preservation of the format. Went with the improvement. However, you're right – if this error is meaningful to and interpreted by the client, this might create an issue. Restored the previous format as much as it was possible.